processor/spanmetricsprocessor/processor.go

```diff
@@ -231,21 +245,40 @@ func (p *processorImp) ConsumeTraces(ctx context.Context, traces pdata.Traces) e
 // writes the raw metrics data into the metrics object.
 func (p *processorImp) buildMetrics() *pdata.Metrics {
 	m := pdata.NewMetrics()
-	ilm := m.ResourceMetrics().AppendEmpty().InstrumentationLibraryMetrics().AppendEmpty()
-	ilm.InstrumentationLibrary().SetName("spanmetricsprocessor")
+	rms := m.ResourceMetrics()
+	for key := range p.resourceAttrList {
+		p.lock.RLock()
+		resourceAttrKey := resourceKey(key)
+		resourceAttributesMap := p.resourceKeyToDimensions[resourceAttrKey]
+
+		// if the service name doesn't exist, we treat it as invalid and do not generate a trace
+		if _, ok := resourceAttributesMap.Get(serviceNameKey); !ok {
+			continue
+		}
+
+		rm := rms.AppendEmpty()
+
+		resourceAttributesMap.Range(func(k string, v pdata.AttributeValue) bool {
```

Comment:
Can you put a comment on this line? I'm not 100% sure what it's doing

Reply:
Added in a comment (which I'll commit soon). Going to resolve this conversation so I know which comments I've addressed but if you are still unsure feel free to re-comment.